src/components/Firestore/DocumentEditor/index.tsx

```diff
@@ -71,8 +76,12 @@ const RootField: React.FC<{
 
   return (
     <>
-      {Object.keys(state).map(field => (
-        <NestedEditor key={field} path={[field]} />
+      {Object.keys(state).map((field, i) => (
```

Comment:
This one can be problematic especially now we key on index. When renaming fields, the iteration order for object keys can change and e.g. the current editor may end up moved to the last one.

I think we may need some special handling to keep things stable (e.g. local caching of existing keys)?

Reply:
Anyway, this is tricky, please change to `key={field}` and dispatch on blur for now (which is at least correct) and let's talk about how to optimize later.

Reply:
Switched back to onBlur... it is still not perfect, but it'll do for EAP. There is an issue where when you enter in a field-name and then click on the type-selector or value-input, technically those elements are re-rendered when the blur-callback is triggered, thus requiring clicking on those elements a second time :(

Reply:
I'll add local-caching as a p1 for next week